datafusion/common/src/column.rs

```diff
@@ -109,21 +109,21 @@ impl Column {
     /// `foo.BAR` would be parsed to a reference to relation `foo`, column name `bar` (lower case)
     /// where `"foo.BAR"` would be parsed to a reference to column named `foo.BAR`
     pub fn from_qualified_name(flat_name: impl Into<String>) -> Self {
```

Comment:
I think we could make this significantly better for the common case (non error) by changing the signature to take `&str` but that would be a breaking change, and I think we should only do it if benchmarking shows it helps.

Most of the time is actually taken parsing the identifier